### ISSUE TITLE
Issues with sounddevice from inside a virtualenv

### ISSUE THREAD

**Issue by illright** [spatialaudio/python-sounddevice#320]
OS: Windows 10
System language: Russian
Python: 3.8.8
Running inside a `virtualenv` (**this is important**) from Pipenv

I've had an issue with the most basic use of `sounddevice` on a certain Windows laptop (I couldn't reproduce it on any other Windows laptop I've got my hands on):
```
Traceback (most recent call last):
  File "threading.py", line 932, in _bootstrap_inner
  File "threading.py", line 870, in run
  File "vpt\sources\device_audio_source.py", line 38, in run_threaded
  File "sounddevice.py", line 274, in rec
  File "sounddevice.py", line 2573, in start_stream
  File "sounddevice.py", line 1415, in _init__ 
  File "sounddevice.py", line 892, in __init__
  File "sounddevice.py", line 2734, in _check 
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xd1 in position 0: invalid continuation byte
```

The above is a traceback from my program, bundled with PyInstaller, but the same issue persists even without threading or PyInstaller, if I just simply run the following line in the Pipenv-created virtualenv Python interpreter:
```powershell
PS> pip install pipenv
PS> pipenv --three
PS> pipenv install sounddevice
PS> pipenv run python
```

```python
>>> import sounddevice as sd
>>> sd.rec(44100, samplerate=44100, channels=2, device=1, blocking=True)
```

What's particularly notable is that if I install `sounddevice` system-wide, outside of Pipenv, the same code works just fine. This leads me to believe that there is some issue with `sounddevice` running inside a virtualenv.

After studying the issues of this repository, I came across https://github.com/jaakkopasanen/Impulcifer/issues/53, where the OP seems to have a similar problem, and they are also running `sounddevice` inside a virtualenv. They have resolved their issue using some setting in Windows, but I wonder if there's a fix on the library side.

**Comment by illright** [comment]
Update: In the issue that I linked, they suggest ticking a box that enables UTF-8 for applications. I did that, and now the error message is different, but unfortunately, the error is still there.

The updated traceback is:
```
Traceback (most recent call last):
  File "threading.py", line 932, in _bootstrap_inner
  File "threading.py", line 870, in run
  File "vpt\sources\device_audio_source.py", line 38, in run_threaded
  File "sounddevice.py", line 274, in rec
  File "sounddevice.py", line 2573, in start_stream
  File "sounddevice.py", line 1415, in _init__ 
  File "sounddevice.py", line 892, in __init__
  File "sounddevice.py", line 2734, in _check 
sounddevice.PortAudioError: Error opening InputStream: Unanticipated host error [PaErrorCode -9999]: 'Системной функции передан неправильный параметр.' [MME error 11]
```

To translate the Russian sentence in the traceback: "A system function received an invalid parameter". What could be causing this?

**Comment by illright** [comment]
Okay, I've uncovered the cause of the issue, it is outside of this library – that laptop was running a corporate version of Windows which was imposing security restrictions and the library couldn't open the audio device. The reason why it worked outside of a virtualenv is because I was launching `python` from Powershell, supposedly a trusted program which had access to hardware.

What have we learned?
1. When an error message contains non-ASCII characters, it can't be decoded and the library dies with an unhelpful error message
2. Windows security policies can be causing MME error 11, there's no way around that, supposedly
